src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java

```diff
@@ -174,6 +180,22 @@ public interface LogStorage {
         }
     }
 
+    /**
+     * Return the primary Log Storage. By default, it's the current implementation.
+     * See {@link org.jenkinsci.plugins.workflow.log.tee.TeeLogStorage} for overriden implementation.
+     */
+    default LogStorage getPrimary() {
```

Comment:
I still think we don't need these methods. We can add a utility method to CPE that checks `logStorage instanceof TeeLogStorage tls { tls.getPrimary() instanceof CPE }`. I think other implementations would never use these methods. If we realize that multiple implementations do need something like it in the future, we can add it later.

Reply:
Removed, see https://github.com/jenkinsci/workflow-api-plugin/pull/417/commits/b2a5025990e5977d165bb63b922cad6ea6bf6363